src/main/scala/controllers/ActionHandler.scala

```diff
@@ -102,27 +102,22 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
   def redirectToPreview(id: ApplicationId, sectionNumber: Int) =
     Redirect(routes.ApplicationPreviewController.previewSection(id, sectionNumber))
 
-  def renderSectionForm(id: ApplicationId,
+  def renderSectionForm(app: ApplicationDetail,
```

Comment:
Well, I am not about having something pure, my concern is that it can be called in sync context from the controller.
I am not into that debate, not sure what is wrong with "private[this]" if you need to re-use a bit of functionality between public methods